packages/flutterfire_cli/lib/src/firebase/firebase_android_writes.dart

```diff
@@ -1051,3 +1077,39 @@ AndroidGradleContents _applyFirebaseAndroidPluginKts({
     gradleSettingsContent: androidGradleSettingsKtsFileContents,
   );
 }
+
+Future<FirebasePubSpecModel> getFirebaseCorePubSpec() async {
+  final pubCacheFolder = _getPubCacheDirectory();
+  final items = pubCacheFolder.listSync();
+  final firebaseCoreDirectory = items
+      .whereType<Directory>()
+      .where((e) => e.path.split('/').last.startsWith('firebase_core-'))
+      .sortedBy((d) => d.path)
+      .last;
+
+  final firebaseCorePubspecFile =
+      pubspecPathForDirectory(firebaseCoreDirectory);
+  final pubSpec = await PubSpec.loadFile(firebaseCorePubspecFile);
+  final unparsedJson = pubSpec.unParsedYaml?['firebase'] as YamlMap?;
+  if (unparsedJson == null) {
+    throw Exception('Could not parse firebase_core pubspec.yaml');
```

Comment:
If parsing fails (i.e. on a version that doesn't have these properties), then we should fall back to the version removed here: https://github.com/invertase/flutterfire_cli/pull/393/files#diff-3dca730df06203f113ee456b77a99080d35449ca1bbe0f8f2293fbb763a42746L59-L75

That way we maintain backwards compatibility as well